src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -1114,5 +1120,15 @@ internal ClusterKey ToClusterKey()
                 _waitQueueSize,
                 _waitQueueTimeout);
         }
+
+        private void Validate()
```

Comment:
Question: is it possible to do this validation on the Cluster level?

Reply:
Name has been changed, so the first comment is slightly out of date? 
I think theoretically we could do it at the cluster level but the spec only calls for preventing it at the MongoClient/connection string level. IMHO doing it at the cluster level is outside the scope of this ticket.

Reply:
>Name has been changed, so the first comment is slightly out of date?

yes

>IMHO doing it at the cluster level is outside the scope of this ticket.

it will allow us to have such validation only once. But I'm ok with the current version.